File: 187-repeated-dna-sequences.cs
public class Solution {
    public IList<string> FindRepeatedDnaSequences(string s) {
        var res = new List<string>();
        if (s.Length < 10) return res;
        var dict = new Dictionary<string, int>();
        for (var i = 0; i <= s.Length - 10; i++) {
            var segment = s.Substring(i, 10);
            dict[segment] = dict.GetValueOrDefault(segment, 0) + 1;
        }

        foreach (var (key, val) in dict)
            if (val >= 2)
                res.Add(key);
        return res;
    }
}

File: 187-repeated-dna-sequences.go
package main

func findRepeatedDnaSequences(s string) []string {
	ret := []string{}
	dict := map[string]int{}

	for i := 0; i <= len(s)-10; i++ {
		segment := s[i : i+10]
		dict[segment]++
		if dict[segment] == 2 {
			ret = append(ret, segment)
		}
	}
	return ret
}
